packages/jest-snapshot/src/index.js

```diff
@@ -0,0 +1,54 @@
+/**
+ * Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+'use strict';
+
+const SnapshotFile = require('./SnapshotFile');
+
+const patchAttr = (attr, state) => {
+  attr.onStart = function(onStart) {
+    return function(context) {
+      const specRunning = context.getFullName();
+      let index = 0;
+      Object.defineProperty(state.specsNextCallCounter, specRunning, {
+        get() {return index++;},
+        enumerable: true,
+      });
+      state.specRunningFullName = specRunning;
+      if (onStart) {
+        onStart(context);
+      }
+    };
+  }(attr.onStart);
+};
+
+const patchJasmine = (jasmine, state) => {
+  jasmine.Spec = (realSpec => {
+    const Spec = function Spec(attr) {
+      patchAttr(attr, state);
+      realSpec.call(this, attr);
+    };
+    Spec.prototype = realSpec.prototype;
+    for (const statics in realSpec) {
+      if (realSpec.hasOwnProperty(statics)) {
+        Spec[statics] = realSpec[statics];
+      }
+    }
+    return Spec;
+  })(jasmine.Spec);
+};
+
+module.exports = {
+  getMatchers: require('./getMatchers'),
+  getSnapshotState: (jasmine, filePath) => {
+    const state = Object.create(null);
+    state.specsNextCallCounter = Object.create(null);
```

Comment:
let's list all the keys out here and initialize them so that we know what we are writing on the state variable.

I also recommend calling it `state.callCount` and `currentSpecName` – it sounds a bit clearer to me :)